.github/actions/wasm-lang-toolchain-setup/action.yml

```diff
@@ -57,15 +62,34 @@ runs:
       run: |
         npm install -g @bytecodealliance/jco
         npm install -g @bytecodealliance/componentize-js
+    - name: clone typescript examples
+      if: ${{ inputs.language == 'typescript' }}
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      with:
+        repository: 'wasmCloud/typescript'
+        path: 'wasmcloud-typescript'
+    - name: copy typescript examples
+      if: ${{ inputs.language == 'typescript' }}
+      shell: bash
+      run: |
+        mkdir -p ./examples/typescript/components
+        cp -r ./wasmcloud-typescript/examples/components/* ./examples/typescript/components/
+        echo "::group::Typescript Examples"
+        ls -l ./examples/typescript/components
+        echo "::endgroup::"
 
+    # All Languages
     # Install wash
     - name: install wash (previous version)
       if: ${{ inputs.wash-version && inputs.wash-version != 'current' }}
-      uses: taiki-e/install-action@437c908c7e5ee18b63a261286cbe5147219f8a39
+      uses: taiki-e/install-action@acd25891978b4cdaebd139d3efef606d26513b14 # v2.47.0
       with:
-        tool: wash-cli@${{ inputs.wash-version }}
+        tool: wash@${{ inputs.wash-version }}
 
     - name: install wash (current)
       if: ${{ inputs.wash-version && inputs.wash-version == 'current' }}
-      shell: bash
-      run: cargo install --path ./crates/wash-cli
+      uses: taiki-e/cache-cargo-install-action@caa6f48d18d42462f9c30df89e2b4f71a42b7c2c # v2.0.1
+      with:
+        tool: wash-cli
+        git: https://github.com/${{ github.repository }}
+        rev: ${{ github.sha }}
```

Comment:
Nifty!

Reply:
It'd be nice if it actually cached the build so that it didn't always build from scratch. I'm going to try with the `Swatinem/rust-cache` action.